File: eCommerce/UI/ecommerce_mobile/lib/model/user_challenge.dart
import 'package:ecommerce_mobile/model/challenge.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_challenge.g.dart';

@JsonSerializable()
class UserChallenge {
  int? userChallengeId;
  int? izazvaniId;
  User? izazvani;
  int? challengeId;
  Challenge? challenge;
  String? statusIzazova;
  DateTime? datumPrihvatanja;
  int? napredak;
  DateTime? vrijemeZavrsetka;

  UserChallenge({
    this.userChallengeId,
    this.izazvaniId,
    this.izazvani,
    this.challengeId,
    this.challenge,
    this.statusIzazova,
    this.datumPrihvatanja,
    this.napredak,
    this.vrijemeZavrsetka,
  });

  factory UserChallenge.fromJson(Map<String, dynamic> json) => _$UserChallengeFromJson(json);

  Map<String, dynamic> toJson() => _$UserChallengeToJson(this);

}

File: eCommerce/eCommerce.Services/Database/UserChallenge.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class UserChallenge
    {
        public int UserChallengeId { get; set; }
        public int IzazvaniId { get; set; }
        public User Izazvani { get; set; }
        public int ChallengeId { get; set; }
        public Challenge Challenge { get; set; }
        public DateTime DatumPrihvatanja { get; set; }
        public StatusIzazov StatusIzazova { get; set; }
        public int Napredak { get; set; }
        public DateTime VrijemeZavrsetka { get; set; }
        public enum StatusIzazov
        {
            Prihvacen=1,
            UToku=2,
            Zavrsen=3
        }
    }
}
